dataflow/src/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -179,6 +194,209 @@ public static Receiver internalReprOf(
+                                        fieldAccessExpression, typeOfId, (VariableElement) ele);
+                        break;
+                }
+        }
+
+        if (receiver == null) {
+            receiver = new Unknown(InternalUtils.typeOf(receiverTree));
+        }
+        return receiver;
+    }
+
+    /**
+     * Returns the implicit receiver of ele.
+     *
+     * <p>Returns either a new ClassName or a new ThisReference depending on whether ele is static
+     * or not. The passed element must be a filed, method, or class.
+     *
+     * @param ele field, method, or class
+     * @return either a new ClassName or a new ThisReference depending on whether ele is static or
+     *     not
+     */
+    public static Receiver internalRepOfImplicitReceiver(Element ele) {
+        TypeMirror enclosingType = ElementUtils.enclosingClass(ele).asType();
+        if (ElementUtils.isStatic(ele)) {
+            return new ClassName(enclosingType);
+        } else {
+            return new ThisReference(enclosingType);
+        }
+    }
+
+    /**
+     * Returns either a new ClassName or ThisReference Receiver for the given tree.
+     *
+     * <p>The Tree should be an expression or a statement that does not have a receiver (or an
+     * implicit receiver). For example, a local variable declaration.
+     *
+     * @param tree an expression or statement that does not have a receiver (or an implicit
+     *     receiver)
+     * @param path TreePath to tree
+     * @param enclosingType type of the enclosing type
+     * @return a new ClassName or ThisReference that is the pseudo receiver of tree
+     */
+    public static Receiver internalRepOfPseudoReceiver(
+            Tree tree, TreePath path, TypeMirror enclosingType) {
+        MethodTree methodTree = TreeUtils.enclosingMethod(path);
+        if (isTreeInStaticScope(tree, methodTree, path)) {
+            return new ClassName(enclosingType);
+        } else {
+            return new ThisReference(enclosingType);
+        }
+    }
+
+    private static boolean isTreeInStaticScope(Tree t, MethodTree enclosingMethod, TreePath path) {
+        if (enclosingMethod != null) {
+            return enclosingMethod.getModifiers().getFlags().contains(Modifier.STATIC);
+        }
+        // no enclosing method, check for static or initailizer block
+        BlockTree block = TreeUtils.enclosingTopLevelBlock(path);
+        if (block != null) {
+            return block.isStatic();
+        }
+
+        // check if its in a variable intializer
+        t = TreeUtils.enclosingVariable(path);
+        if (t != null) {
+            return ((VariableTree) t).getModifiers().getFlags().contains((Modifier.STATIC));
+        }
+        ClassTree classTree = TreeUtils.enclosingClass(path);
+        if (classTree != null) {
+            return classTree.getModifiers().getFlags().contains((Modifier.STATIC));
+        }
+        return false;
+    }
+
+    private static Receiver internalRepOfMemberSelect(
```

Comment:
Calling internalRepOfMemberSelect is equivalent to calling internalRepOf, so it would be confusing if it was public.  User would have to figure out which one to call. Calling  internalRepOfPseudoReceiver is not equivalent to calling internalRepOf.